src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -106,15 +112,25 @@ private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, IPro
 
         private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, CancellationToken cancellationToken)
         {
-            var coverageUtility = new CoverageFileUtility();
-
-            var coverageData = await coverageUtility.MergeCoverageFilesAsync(
-                    files,
-                    cancellationToken).ConfigureAwait(false);
-
-            coverageUtility.WriteCoverageFile(files[0], coverageData);
-
-            foreach(var file in files.Skip(1))
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var assemblyPath = Path.Combine(Path.GetDirectoryName(typeof(CodeCoverageDataAttachmentsHandler).GetTypeInfo().Assembly.GetAssemblyLocation()), CodeCoverageCoreLibNetAssemblyName + ".dll");
+            
+            // Get assembly, type and methods
+            Assembly assembly = new PlatformAssemblyLoadContext().LoadAssemblyFromPath(assemblyPath);
+            var classType = assembly.GetType($"{CodeCoverageCoreLibNetAssemblyName}.{CoverageFileUtilityTypeName}");
+            var classInstance = Activator.CreateInstance(classType);
+            var mergeMethodInfo = classType?.GetMethod(MergeMethodName, new[] { typeof(IList<string>), typeof(CancellationToken) });
+            var writeMethodInfo = classType?.GetMethod(WriteMethodName);
+            
+            // Invoke methods
+            var task = (Task)mergeMethodInfo.Invoke(classInstance, new object[] { files, cancellationToken });
```

Comment:
```suggestion
            var task = mergeMethodInfo.Invoke(classInstance, new object[] { files, cancellationToken }) as Task;
```

Reply:
This might change behavior, code is cherrypicked from master. We should do this on master IMO.